google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -140,21 +140,31 @@ public DocumentReference document(@Nonnull String documentPath) {
   @Override
   public ApiFuture<List<DocumentSnapshot>> getAll(
       @Nonnull DocumentReference... documentReferences) {
-    return this.getAll(documentReferences, null, null);
+    return this.getAll(documentReferences, null, null, null);
   }
 
   @Nonnull
   @Override
   public ApiFuture<List<DocumentSnapshot>> getAll(
       @Nonnull DocumentReference[] documentReferences, @Nullable FieldMask fieldMask) {
-    return this.getAll(documentReferences, fieldMask, null);
+    return this.getAll(documentReferences, fieldMask, null, null);
+  }
+
+  @Nonnull
+  @Override
+  public void getAll(
+      @Nonnull DocumentReference[] documentReferences,
+      @Nullable FieldMask fieldMask,
+      final ApiStreamObserver<DocumentSnapshot> responseObserver) {
+    this.getAll(documentReferences, fieldMask, null, responseObserver);
   }
 
   /** Internal getAll() method that accepts an optional transaction id. */
   ApiFuture<List<DocumentSnapshot>> getAll(
       final DocumentReference[] documentReferences,
       @Nullable FieldMask fieldMask,
-      @Nullable ByteString transactionId) {
+      @Nullable ByteString transactionId,
+      @Nullable final ApiStreamObserver apiStreamObserver) {
```

Comment:
Do you think we could have one method that adds the ApiStreamObserver, and one that turns it into an ApiFuture?

```
void getAll(
      final DocumentReference[] documentReferences,
      @Nullable FieldMask fieldMask,
      @Nullable ByteString transactionId,
      @Nullable final ApiStreamObserver apiStreamObserver) { ... }

ApiFuture<List<DocumentSnapshot>> getAll(
      final DocumentReference[] documentReferences,
      @Nullable FieldMask fieldMask,
      @Nullable ByteString transactionId) { 
   List results....
   getAll(..., new ApiStreamObserver() -> {
      result.add();
    });
    // Sort and return task
 }
```

Reply:
done